tools/cli.js

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env node
+
+'use strict';
+
+// TODO: support jstp://server and jstps://server
+
+const jstp = require('.');
+const readline = require('readline');
+
+const commandProcessor = {};
+const lineProcessor = {};
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+  completer
+});
+
+const log = msg => {
+  const userInput = rl.line;
+  if (userInput) rl.clearLine();
+  console.log(msg);
+  rl.write('\n');
+  if (userInput) rl.write(userInput);
+};
+
+rl.on('line', (line) => {
+  const [type, leftover] = _split(line.trim(), ' ', 1);
+  if (!type) {
+    return rl.prompt(true);
+  }
+  const processor = lineProcessor[type];
+  if (!processor) {
+    log(`Unknown command ${type}`);
+  } else {
+    processor(leftover, (err, result) => {
+      if (err) return log(`${err.name} occurred: ${err.message}`);
+      log(result);
+    });
+    rl.prompt(true);
+  }
+});
+
+rl.on('SIGINT', () => {
+  rl.close();
+  process.exit();
+});
+
+rl.on('close', () => {
+  process.exit();
+});
+
+function completer(line) {
+  return [[], line];
+}
+
+const state = {
+  client: null,
+  connection: null
+};
+
+commandProcessor.call = (interfaceName, methodName, args, callback) => {
+  if (state.client === null) return callback(new Error('Not connected'));
```

Comment:
Not sure about whether the braces would help readability since the next line is quite long too.